### PR TITLE
Relocate and Expose Application and Handler type.

### PR DESCRIPTION
**Pull request by danilvpetrov**
#### What change does this introduce?

This PR relocates the `application` and `handler` types to `internal/entity` package. It also exposes these types so they can be reused in the code of this repository.

#### What issues does this relate to?

Partially relates to #104.

#### Why make this change?

This change is required so that `Application` and `Handler` types can be reused throughout the code of this repository.

#### Is there anything you are unsure about?

Not really.

### CODE REVIEW

File: internal/entity/entity_test.go [hunk 1]
```diff
@@ -11,10 +11,10 @@ import (
 )
 
 var _ = Describe("type application", func() {
```
Comment (jmalloc) [inline internal/entity/entity_test.go:13]:
```suggestion
var _ = Describe("type Application", func() {
```

File: internal/entity/entity_test.go [hunk 2]
```diff
@@ -33,10 +33,10 @@ var _ = Describe("type application", func() {
 })
 
 var _ = Describe("type handler", func() {
```
Comment (jmalloc) [inline internal/entity/entity_test.go:35]:
```suggestion
var _ = Describe("type Handler", func() {
```

**Review by jmalloc — CHANGES_REQUESTED** [review]
Is this just missing its `ginkgo_test.go` file?

**Review by koden-km — APPROVED** [review]

File: internal/entity/entity.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+	"github.com/dogmatiq/configkit"
+)
+
+// Application is an implementation of config.Application.
+type Application struct {
+	IdentityValue configkit.Identity
+	TypeNameValue string
+	Messages      configkit.EntityMessageNames
+	HandlerSet    configkit.HandlerSet
```
Comment (jmalloc) [inline internal/entity/entity.go:14]:
```suggestion
	MessageNameValue      configkit.EntityMessageNames
	HandlersValue    configkit.HandlerSet
```

Should we just make these all use `[method name] + Value`, like this?

Comment (danilvpetrov) [reply internal/entity/entity.go:14]: Yeah I was thinking about it, but then foolishly made exceptions for `Messages`, etc. Let me change everything to look uniform.

Comment (danilvpetrov) [reply internal/entity/entity.go]: Ok that should be done.

**Comment by danilvpetrov** [comment]
Since we taken this two types out to its individual package, coverage started suffering a bit, do we want to add some tests to address that?

**Comment by jmalloc** [comment]
Yeah, definitely.

**Comment by danilvpetrov** [comment]
Ok, looks like we're back on track with the coverage, can you take another look?

**Review by jmalloc — APPROVED** [review]

**Pull request merged**